### ISSUE TITLE
the drop measurement reappears and the data is old

### ISSUE THREAD

**Issue by EmptyRabbit** [influxdata/influxdb#10263]
__System info:__ 
influxdb version:1.5.2
operating system:ubuntu
tools:iterm,pycharm
language:python3.6

__Steps to reproduce:__

1.use python-influxdb  DataFrameClient.write_points write data to measurement 'TS121'
2. drop measurement TS121
3.use python-influxdb  DataFrameClient.write_points write same data to measurement 'TS121' again
4.drop measurement TS121
5.use python-influxdb  DataFrameClient.write_points write other data to measurement 'TS121' again
6.drop measurement TS121
7.select * from TS121 limit 1

__Expected behavior:__ 
after step 7
I expected nothing would be printed

__Actual behavior:__
after step 7
the screen show TS121 has data I wrote in step 1&3 not data in step 5
using DataFrameClient.query also could query the data in step 1&3

__Additional info:__ 
don't know much details.but drop other measurement is ok

**Comment by e-dard** [comment]
Possibly a duplicate of #9694

**Comment by russorat** [comment]
@e-dard can this be closed?

**Comment by EmptyRabbit** [comment]
Recently I get the similar bug:
1、measuremnt A has fields a,b,c
2、drop measurement A
3、write b,c into measurement A
4、expected is field only b,c but I get a,b,c .field a data is totaly empty

1、it's an old database
2、index-version = "inmem"
3、bug still after restart influxdb
4、write after drop
5、my version is 1.6.2 not try 1.7.2
@zeebo

**Comment by EmptyRabbit** [comment]
> 我想更多地了解一下这个：
> 
> 1. 这是一个新鲜的数据库吗？
> 2. 您的分片使用了哪些索引？
> 3. 重启后问题是否仍然存在？
> 4. 您是否在放弃测量的同时写入测量？
> 5. 问题是否仍然发生在1.7.1版本上？
> 
> 要收集分片索引类型，您可以导航到分片文件夹并检查它是否有`index`文件夹，或运行此命令：
> 
> ```
> influx -database <database> -execute 'show stats' | grep indexType | sort | uniq
> ```

I meet the situation again .is this a bug in 1.6.2?

最近我得到了类似的错误：
1，measuremnt A有字段a，b，c 
2，丢弃测量A 
3，写入b，c到测量A 
4，预期只有字段b，c但我得到a，b，c。字段数据完全为空

1，它是一个旧的数据库
2，index-version =“inmem” 
3，bug仍然在重启后涌入流媒体
4，写下降后
5，我的版本是1.6.2不试试1.7.2

> I'd love to have some more details on this:
> 
> 1. Is this on a fresh database?
> 2. What indexes are being used for your shards?
> 3. Does the issue persist after a restart?
> 4. Are you writing to the measurement the same time you're dropping it?
> 5. Does the issue still happen on version 1.7.1?
> 
> To collect shard index types, you can either navigate to the shard folder and check if it has an `index` folder, or run this command:
> 
> ```
> influx -database <database> -execute 'show stats' | grep indexType | sort | uniq
> ```

I meet the situation again .is this a bug in 1.6.2?
Recently I get the similar bug:
1、measuremnt A has fields a,b,c
2、drop measurement A
3、write b,c into measurement A
4、expected is field only b,c but I get a,b,c .field a data is totaly empty

1、it's an old database
2、index-version = "inmem"
3、bug still after restart influxdb
4、write after drop
5、my version is 1.6.2 not try 1.7.2

**Comment by EmptyRabbit** [comment]
> I'd love to have some more details on this:
> 
> 1. Is this on a fresh database?
> 2. What indexes are being used for your shards?
> 3. Does the issue persist after a restart?
> 4. Are you writing to the measurement the same time you're dropping it?
> 5. Does the issue still happen on version 1.7.1?
> 
> To collect shard index types, you can either navigate to the shard folder and check if it has an `index` folder, or run this command:
> 
> ```
> influx -database <database> -execute 'show stats' | grep indexType | sort | uniq
> ```

I meet the situation again ,is this a bug in 1.6.2?

Recently I get the similar bug:
1、measuremnt A has fields a,b,c
2、drop measurement A
3、write b,c into measurement A
4、expected is field only b,c but I get a,b,c .field a data is totaly empty

1、it's an old database
2、index-version = "inmem"
3、bug still after restart influxdb
4、write after drop
5、my version is 1.6.2 not try 1.7.2

**Comment by e-dard** [comment]
@EmptyRabbit I think I fixed this in https://github.com/influxdata/influxdb/pull/14266. It will be available in 1.8 or 1.7.9 (if there is a 1.7.9).
